File: src/main/java/com/doestheohavethisbook/BooksController.java
package com.doestheohavethisbook;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/books")
public class BooksController {
    @Autowired
    private BookService bookService = new BookServiceImpl();

    @CrossOrigin
    @GetMapping
    public ResponseEntity<List<Book>> getAllBooks() {
        return new ResponseEntity<>(bookService.getAllBooks(), HttpStatus.OK);
    }
}

File: src/main/java/com/doestheohavethisbook/BooksController.kt
package com.doestheohavethisbook

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/books")
class BooksController {
    @Autowired
    lateinit var bookService: BookService

    @CrossOrigin
    @GetMapping
    fun getAllBooks(): ResponseEntity<List<Book>> {
        return ResponseEntity(bookService.getAllBooks(), HttpStatus.OK)
    }
}
